src/arithmetic/algebraic_expression/algebraic_expression_eval.c

```diff
@@ -11,6 +11,58 @@
 // Forward declarations
 GrB_Matrix _AlgebraicExpression_Eval(const AlgebraicExpression *exp, GrB_Matrix res);
 
+//
+static GrB_Matrix _Eval_Extract(const AlgebraicExpression *exp, GrB_Matrix res) {
+	assert(exp &&
+		   AlgebraicExpression_ChildCount(exp) > 1 &&
+		   AlgebraicExpression_OperationCount(exp, AL_EXP_MUL) == 1);
+
+	GrB_Vector V;
+	GrB_Matrix B;
+	GrB_Info info;
+	GrB_Index nrows;
+	GrB_Index ncols;
+	GrB_Index nvals;
+	GrB_Descriptor desc;
+	AlgebraicExpression *left = CHILD_AT(exp, 0);
+	AlgebraicExpression *right = CHILD_AT(exp, 1);
+
+	info = GrB_Descriptor_new(&desc);  // Descriptor used for transposing operands.
+
+	assert(left->type == AL_OPERAND && left->operand.type == AL_OPERAND_VECTOR);
+	V = left->operand.vector;
+
+	if(right->type == AL_OPERATION) {
+		assert(right->operation.op == AL_EXP_TRANSPOSE);
+		info = GrB_Descriptor_set(desc, GrB_INP0, GrB_TRAN);
+		right = CHILD_AT(right, 0);
+	}
+	B = right->operand.matrix;
+
+	info = GrB_Vector_nvals(&nvals, V);
+	GrB_Index indicies[nvals];
+
+	// Extract indicies.
+	info = GrB_Vector_extractTuples_UINT64(GrB_NULL, indicies, &nvals, V);
+
+	if(B == IDENTITY_MATRIX) {
+		/* v * I
+		 * as I doesn't really exist, we'll manually populate `res`.
+		 * each entry x in v, v[x] will extract row v[x] from I. */
+		for(uint i = 0; i < nvals; i++) GrB_Matrix_setElement_BOOL(res, true, i, indicies[i]);
```

Comment:
I'm a little confused here. We've not stored the row indices associated with the values of `indicies`, so can we safely populate 0..nvals? What if there was azero value in the middle of the vector? Am I misunderstanding this section?

Reply:
I see now that for a filter vector, there is never a non-zero value in the middle! So this is fine for the current setup, but might not be extensible?